graph/src/blockchain/mod.rs

```diff
@@ -305,3 +310,73 @@ pub trait RuntimeAdapter<C: Blockchain>: Send + Sync {
 pub trait NodeCapabilities<C: Blockchain> {
     fn from_data_sources(data_sources: &[C::DataSource]) -> Self;
 }
+
+/// Blockchain technologies supported by Graph Node.
+#[derive(Copy, Clone, PartialEq, Eq, Hash)]
+pub enum BlockchainKind {
+    /// Ethereum itself or chains that are compatible.
+    Ethereum,
+}
+
+impl fmt::Display for BlockchainKind {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        let value = match self {
+            BlockchainKind::Ethereum => "ethereum",
+        };
+        write!(f, "{}", value)
+    }
+}
+
+impl FromStr for BlockchainKind {
+    type Err = Error;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "ethereum" => Ok(BlockchainKind::Ethereum),
+            _ => Err(anyhow!("unknown blockchain kind {}", s)),
+        }
+    }
+}
```

Comment:
How do you feel about delegating this to `serde::plain`, just like we did for [subgraph features](https://github.com/graphprotocol/graph-node/blob/master/graph/src/data/subgraph/features.rs#L36-L51)?

While the current implementation works flawlessly, if we use `serde` here we wouldn't need to write any more code when introducing new variants.

Reply:
I don't have much of a preference so I'll keep it as it is.